libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -602,13 +602,15 @@ bool AP_Arming::gps_checks(bool report)
         }
 
         // check AHRS and GPS are within 10m of each other
-        const Location gps_loc = gps.location();
-        Location ahrs_loc;
-        if (AP::ahrs().get_location(ahrs_loc)) {
-            const float distance = gps_loc.get_distance(ahrs_loc);
-            if (distance > AP_ARMING_AHRS_GPS_ERROR_MAX) {
-                check_failed(ARMING_CHECK_GPS, report, "GPS and AHRS differ by %4.1fm", (double)distance);
-                return false;
+        if (gps.num_sensors() > 0) {
```

Comment:
This obviously loosens the checks, which is good.

It doesn't address one of the issues directly - where we have GPS sensors, but the EKF is currently configured not to use them (e.g. via ek3-source-sets).  In this situation the user may or may not want to arm (depending on their use case).  If that use case is transitioning from a GPS-denied env where it is being armed to a GPS environment then they want most of the checks in this function but not this one.

Reply:
@peterbarker,

Txs very much for the review and approval.

I don't quite understand the last sentence but I think you're saying we still have issues when the GPS is configured but the vehicle is being armed in a non-GPS environment.  I totally agree and I hope we can fix those in a follow-up PR at some point.